### PR TITLE
OCLOMRS-252: Add all bulkconcepts fix

### PR DESCRIPTION
**Pull request by Michyjones**
# JIRA TICKET NAME:
[OCLOMRS-252: Add all bulk concepts fix](https://issues.openmrs.org/browse/OCLOMRS-252)

# Summary:
Users can be able to add all bulk concepts at once.

### CODE REVIEW

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx [hunk 1]
```diff
@@ -102,13 +106,7 @@ export class DictionaryConcepts extends Component {
...
     const { typeName, collectionName, searchInput } = this.state;
     if (type === 'checkbox' && filterType === 'source') {
-      this.props.filterBySource(
-        inputName,
-        this.state.type,
-        typeName,
-        collectionName,
-        searchInput,
-      );
+      this.props.filterBySource(inputName, this.state.type, typeName, collectionName, searchInput);
```
Comment (dondrzzy) [inline src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: Is this a formatting change?

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ export class DictionaryConcepts extends Component {
     this.state = {
       conceptsCount: this.props.totalConceptCount,
       searchInput: '',
-      conceptLimit: 10,
+      conceptLimit: 20,
```
Comment (dondrzzy) [inline src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: Why are you increasing the concept limit?

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx [hunk 3]
```diff
@@ -71,6 +71,10 @@ export class DictionaryConcepts extends Component {
         params: { collectionName, type, typeName },
       },
     } = nextProps;
+    localStorage.setItem('dictionaryId', this.props.match.params.collectionName);
```
Comment (dondrzzy) [inline src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: Why are you storing all these items in the local storage?

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx [hunk 4]
```diff
@@ -137,26 +135,24 @@ export class DictionaryConcepts extends Component {
...
     }));
   }
 
   checkMembershipStatus(username) {
     const {
       match: {
-        params: {
-          type, typeName,
-        },
+        params: { type, typeName },
```
Comment (dondrzzy) [inline src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: Is this a formatting change?

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 2 above)
Comment (Michyjones) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @dondrzzy Pagination limit should be 20 items according to the mockup. That is why I changed that limit.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 1 above)
Comment (Michyjones) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @dondrzzy yes, It's formatting change.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 3 above)
Comment (Michyjones) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @dondrzzy  initially there was localStorage.getItem but it was not set. That is why I decided to set the Item fast.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 4 above)
Comment (Michyjones) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @dondrzzy yes, It's formatting change.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx [hunk 5]
```diff
@@ -167,11 +163,17 @@ export class DictionaryConcepts extends Component {
...
+      data,
+      type,
+      this.props.match.params.typeName,
+      collectionName,
+    );
+  };
 
-  handleShowDelete = (url) => { this.setState({ data: { references: [url] }, versionUrl: url }); };
+  handleShowDelete = (url) => {
+    this.setState({ data: { references: [url] }, versionUrl: url });
```
Comment (alexmochu) [inline src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: I have noticed quite a number of formatting changes, don't you think that would be best handled by a ticket of of its own?

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 4 above)
Comment (dondrzzy) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: Those should be best be handled in a different PR.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 1 above)
Comment (dondrzzy) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: These should be addressed in a different PR.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 3 above)
Comment (dondrzzy) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: > @dondrzzy initially there was localStorage.getItem but it was not set. That is why I decided to set the Item fast.

I believe you meant `first`. That is okay then. Good work

**Comment by dondrzzy** [comment]
Please update the ticket in response to the comment left there. @Michyjones

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 1 above)
Comment (Michyjones) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @dondrzzy   I have reverted all the formatting changes.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 5 above)
Comment (Michyjones) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @alexmochu  I have reverted all the formatting changes.

**Comment by Michyjones** [comment]
@dondrzzy I have responded to the comments and made the requested changes

**Comment by dkayiwa** [comment]
@Michyjones why has this taken close to a week without closing?

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 5 above)
Comment (alexmochu) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @Michyjones No you have not, I can still see the changes.

**Comment by dkayiwa** [comment]
@Michyjones ping

**Comment by Michyjones** [comment]
@dkayiwa I have not being paying attention to this. I did not get any notification when you posted this comment.

**Comment by Michyjones** [comment]
@dondrzzy and @alexmochu  I have reverted all the formatting changes. Could you take a look at it

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 5 above)
Comment (alexmochu) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: Can you please share the ticket that will handle the linting errors above.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 2 above)
Comment (alexmochu) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: But it was not part of the scope. Why not create a separate ticket for this.

File: src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx (hunk 5 above)
Comment (Michyjones) [reply src/components/dictionaryConcepts/containers/DictionaryConcepts.jsx]: @alexmochu I realised that I had configured my eslint wrongly. There are no more highlighting errors.

**Comment by dondrzzy** [comment]
@Michyjones, Endeavour to always be the last to comment on every review so that reviewers can know whether you have addressed the issue. Good work.

**Comment by Michyjones** [comment]
@dondrzzy Thanks for that reminder. I will follow that approach to make sure I have address to every comment on the PR thanks again.

**Pull request merged**